src/cart/crt_context.c

```diff
@@ -395,6 +395,13 @@ crt_ctx_epi_abort(d_list_t *rlink, void *arg)
 	D_ASSERT(rlink != NULL);
 	D_ASSERT(arg != NULL);
 	epi = epi_link2ptr(rlink);
+
+	/*
+	 * DAOS-7306: This mutex is needed in order to avoid double
+	 * completions that would happen otherwise. safe list processing
+	 * is not sufficient to avoid the race
+	 */
+	D_MUTEX_LOCK(&epi->epi_mutex);
```

Comment:
ok, looks fine to add epi_mutex here, to protect accessing of "epi_req_waitq/epi_req_q".